### PR TITLE
refactor southpark's code

### PR DESCRIPTION
**Pull request by YueLiu-jina**

### CODE REVIEW

File: southpark-search/app.py [hunk 1]
```diff
@@ -35,29 +28,66 @@ def print_topk(resp, word):
...
+    with f:
+        while True:
+            text = input('please type a sentence: ')
+            if not text:
+                break
+            ppr = lambda x: print_topk(x, text)
+            f.search_lines(lines=[text, ], output_fn=ppr, topk=top_k)
+
+def query_restful(num_docs):
+    config(num_docs,mode = 'search')
```
Comment (JoanFM) [inline southpark-search/app.py:60]: `config(num_docs, mode='search)`

Comment (JoanFM) [reply southpark-search/app.py:60]: Style comment

File: southpark-search/app.py [hunk 2]
```diff
@@ -35,29 +28,66 @@ def print_topk(resp, word):
...
+    os.environ['SHARDS'] = os.environ.get('SHARDS', str(2))
+    os.environ['DATA_DIR'] = os.environ.get('DATA_DIR', '/tmp/jina/southpark')
+    os.environ['DATA_FILE'] = os.environ.get('DATA_FILE', 'character-lines.csv')
+    os.environ['TMP_WORKSPACE'] = os.environ.get('TMP_WORKSPACE', get_random_ws(os.environ['DATA_DIR']))
+    os.environ['MAX_NUM_DOCS'] = os.environ.get('MAX_NUM_DOCS', str(num_docs))
+    os.environ['JINA_PORT'] = os.environ.get('JINA_PORT', str(45678))
+
+
+def index(num_docs):
+    config(num_docs,mode = 'index')
```
Comment (JoanFM) [inline southpark-search/app.py:42]: Same here

File: southpark-search/app.py [hunk 3]
```diff
@@ -35,29 +28,66 @@ def print_topk(resp, word):
...
+def query_restful(num_docs):
+    config(num_docs,mode = 'search')
+    f = Flow().load_config('flow-query.yml')
+    f.use_rest_gateway()
+    with f:
+        f.block()
+
+
+def dryrun(num_docs):
+    config(num_docs,mode = 'dryrun')
```
Comment (JoanFM) [inline southpark-search/app.py:68]: Same

File: southpark-search/app.py [hunk 4]
```diff
@@ -35,29 +28,66 @@ def print_topk(resp, word):
...
+
+def index(num_docs):
+    config(num_docs,mode = 'index')
+    data_path = os.path.join(os.environ['DATA_DIR'], os.environ['DATA_FILE'])
+    f = Flow().load_config('flow-index.yml')
+    with f:
+        f.index_lines(filepath=data_path, batch_size=8, size=int(os.environ['MAX_NUM_DOCS']))
+
+def query(num_docs, top_k):
+    config(num_docs, mode = 'search')
```
Comment (JoanFM) [inline southpark-search/app.py:49]: Same

File: southpark-search/app.py (hunk 1 above)
Comment (YueLiu-jina) [reply southpark-search/app.py:60]: oh yes, thanks for reminding

**Pull request closed without merging**